sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -53,24 +53,31 @@ import org.apache.spark.sql.util.SchemaUtils
  * are identical to the ones defined in the source table.
  *
  * The CatalogTable attributes copied from the source table are storage(inputFormat, outputFormat,
- * serde, compressed, properties), schema, provider, partitionColumnNames, bucketSpec.
+ * serde, compressed, properties), schema, provider, partitionColumnNames, bucketSpec by default.
  *
- * Use "CREATE TABLE t1 LIKE t2 USING file_format"
- * to specify new file format for t1 from a data source table t2.
+ * Use "CREATE TABLE t1 LIKE t2 USING file_format" to specify new provider for t1.
+ * For Hive compatibility, use "CREATE TABLE t1 LIKE t2 STORED AS hiveFormat"
+ * to specify new file storage format (inputFormat, outputFormat, serde) for t1.
  *
  * The syntax of using this command in SQL is:
  * {{{
  *   CREATE TABLE [IF NOT EXISTS] [db_name.]table_name
- *   LIKE [other_db_name.]existing_table_name [USING provider] [locationSpec]
+ *   LIKE [other_db_name.]existing_table_name [USING provider | STORED AS hiveFormat]
+ *   [locationSpec] [TBLPROPERTIES (property_name=property_value, ...)]
  * }}}
  */
 case class CreateTableLikeCommand(
     targetTable: TableIdentifier,
     sourceTable: TableIdentifier,
     provider: Option[String],
+    hiveFormat: Option[CatalogStorageFormat],
     location: Option[String],
```

Comment:
`CatalogStorageFormat` has the location field, shall we remove this parameter?

For data source table, we can create `CatalogStorageFormat.empty.copy(locationUri = ...)` in the parser

Reply:
> For data source table, we can create CatalogStorageFormat.empty.copy(locationUri = ...) in the parser

If that, below assertion should be removed and 
`hiveFormat: Option[CatalogStorageFormat]` should be changed to 
`fileFormat: CatalogStorageFormat` and its name can not be called hiveFormat.
It's a little confuse. Is it worth to do that?

Reply:
That's why I suggest moving the below check to the parser.

Reply:
It increases the complex of code. If we always pass a `CatalogStorageFormat` in `CreateTableLikeCommand `. How to know when use the passed CatalogStorageFormat and when use the `sourceTableDesc.storage`?
Current we use `hiveFormat match { case Some(f) => case None =>}` to determinate it.

Reply:
For example, if the `sourceTableDesc.storage` is ORC format. And the CTL command contains a LOCATION but without a STORED AS part. So the `CatalogStorageFormat.empty.copy(locationUri = ...)` will passes to `CreateTableLikeCommand`. In this case we should use the same format as source table's (ORC). So code will change to
```scala
    val newStorage = sourceTableDesc.storage.copy(
        locationUri = if (fileFormat.locationUri.isDefined) {
          fileFormat.locationUri
        } else {
          sourceTableDesc.storage.locationUri
        },
        inputFormat = if (fileFormat.inputFormat.isDefined) {
          fileFormat.inputFormat
        } else {
          sourceTableDesc.storage.inputFormat
        },
        outputFormat = if (fileFormat.outputFormat.isDefined) {
          fileFormat.outputFormat
        } else {
          sourceTableDesc.storage.outputFormat
        },
        serde = if (fileFormat.serde.isDefined) {
          fileFormat.serde
        } else {
          sourceTableDesc.storage.serde
        })
```